### ISSUE TITLE
`pure_return` does not return a `pure_wp`

### ISSUE THREAD

**Issue by dominique-unruh** [FStarLang/FStar#2310]
To my understanding, `pure_return` should return a `pure_wp`. However, this is not the case:
```
assume val test : pure_wp unit -> unit

// Fails to type check because pure_return does not return a pure_wp
let _ = test (pure_return unit ())
```
In fact, I think the definition of the `PURE` effect in `Prims` actually shouldn't even type check with this definition of `pure_return`. (I guess the type checking is partially disabled there because `PURE` is built in.)

I think the definition of `pure_wp` should be something like this:
```
open FStar.Tactics

unfold
let pure_return (a: Type) (x: a) : (pure_wp a) 
  by (norm [delta_only ["Prims.pure_wp_monotonic"]]) =
  fun (p: pure_post a) ->
  forall (return_val: a). return_val == x ==> p return_val
```

**Comment by aseemr** [comment]
Interestingly I also noticed this yesterday in another context. As you point out, this is due to the fact that when `PURE` is being defined, we don't have all the scaffolding setup, the prelude of `prims` until the definition of `GTot` is trusted.

I worked around by writing `FStar.Monotonic.Pure.as_pure_wp (pure_return unit ())`. But this is also suboptimal since the client has to prove the monotonicity predicate.

I am trying to fix it by defining `pure_return` etc. in `FStar.Pervasives`. At that point we verify everything, so we can return the `pure_wp` type, plus `Pervasives` is also automatically opened in the client namespace, so the clients should be able to write:


```
assume val test : pure_wp unit -> unit
let _ = test (pure_return unit ())
```
